### PR TITLE
chore(deps): bump arrow-udf-js and rquickjs

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

to fix the stuck issue https://github.com/DelSkayn/rquickjs/issues/268

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -5097,12 +5097,12 @@ dependencies = [
  "anyhow",
  "apache-avro 0.17.0",
  "arrow-arith 50.0.0",
- "arrow-array 50.0.0",
- "arrow-buffer 50.0.0",
+ "arrow-array 48.0.1",
+ "arrow-buffer 48.0.1",
```
Comment (wangrunji0408) [inline Cargo.lock:5101]: Why icelake is affected? I only ran `cargo update -p arrow-udf-js`

File: Cargo.lock [hunk 2]
```diff
@@ -5097,12 +5097,12 @@ dependencies = [
  "anyhow",
  "apache-avro 0.17.0",
  "arrow-arith 50.0.0",
- "arrow-array 50.0.0",
- "arrow-buffer 50.0.0",
+ "arrow-array 48.0.1",
```
Comment (BugenZhao) [inline Cargo.lock:5100]: 👀

**Review by BugenZhao — APPROVED** [review]
Rest LGTM

File: Cargo.lock (hunk 1 above)
Comment (BugenZhao) [reply Cargo.lock:5101]: `icelake` specifies `arrow >=48`. Rare case. 😄

https://github.com/icelake-io/icelake/blob/54fd72fbd1dd8c592f05eeeb79223c8a6a33c297/Cargo.toml#L15-L23

Comment (xxchan) [reply Cargo.lock:5101]: Because icelake depends on `>=48`, and we depends only multiple versions of arrow. So this is indeed valid...

Comment (BugenZhao) [reply Cargo.lock:5101]: cc @liurenjie1024 `>=48` seems not to always work. Can we specify a fixed major version to conform the semantics versioning?

Comment (wangrunji0408) [reply Cargo.lock]: Manually reverted this change.

**Pull request merged**